app/src/org/commcare/CommCareApplication.java

```diff
@@ -1476,4 +1478,7 @@ public boolean checkPendingBuildRefresh() {
         return false;
     }
 
+    public PrototypeFactory getPrototypeFactory(Context c) {

```

Comment:
It does feel wrong and there are various tools to get around it, like dependency injection frameworks.

I would like to revisit this at some point, but for now it is the best way I could come up with.

And yeah, overriding static methods isn't a thing in Java.